src/faultlog/deconfig_records.cpp

```diff
@@ -85,17 +85,27 @@ int DeconfigRecords::getCount(const GuardRecords& guardRecords)
     DeconfigDataList deconfigList;
     pdbg_target_traverse(nullptr, getDeconfigTargets, &deconfigList);
     DeconfigDataList onlyDeconfigList;
+    // consider only those targets that are not part of guard list
     for (const auto& target : deconfigList.targetList)
     {
         ATTR_PHYS_DEV_PATH_Type attrPhyDevPath;
         if (!DT_GET_PROP(ATTR_PHYS_DEV_PATH, target, attrPhyDevPath))
         {
-            // consider only those targets that are not part of guard list
-            if (std::find(pathList.begin(), pathList.end(), attrPhyDevPath) !=
-                pathList.end())
+            auto it = std::find_if(
+                pathList.begin(), pathList.end(),
+                [attrPhyDevPath](const auto& path) {
+                    return strncmp(attrPhyDevPath, path.c_str(),
```

Comment:
we can get rid of strncmp right?

Reply:
the ATTR_PHYS_DEV_PATH_Type is a character array so we cannot use std::string == operator, the defect was due to my usage of std::string == operator, now using strncmp

Reply:
Means it non-null terminated char array? then how pushing it as a string is working?

Reply:
it has some special characters when I print both the strings are printing same but when i compare if manual guard record is "physical:sys-0/node-0/proc-1" it should be ignored as part of deconfig records or else there will be duplicate record but it is not ignoring if i use std::string comparision.

Reply:
now building string size of character array